src/scancode/cli.py

```diff
@@ -382,6 +382,9 @@ def scan_event(item):
                     return ''
                 if item:
                     _scan_success, _scanned_path = item
+                    if len(_scanned_path)>20:
+                    	current_filename, current_file_extension = os.path.splitext(_scanned_path)
+                    	_scanned_path = current_filename[:3] + '...' + current_filename[-3:] + current_file_extension
```

Comment:
keeping only three characters at start and end does not sound right.
IMHO you should also isolate this in its own function so it can be tested.